### DISCUSSION TITLE
New Feature / Plugin - Historical Test Results Visualization

### DISCUSSION THREAD

**Discussion by captainp1an3t** [ctrf-io/ctrf#16] [Reporters]
I think standardizing on a format is awesome. The benefits of setting the output directly in GitHub has been immediately recognized on my team. I had hopes of now extending the benefits to visualizing historical data (ie: graphs of some sort) that would indicate how tests have performed over a period of time, how long certain tests are taking, etc..

see 
![IMG_1472](https://github.com/user-attachments/assets/cc09fbba-4f19-45d0-b3e7-3cf177259087)

Note:
After looking up examples of graphs I actually stumbled upon a potential partnership for crtf 

https://allurereport.org

It seems they’re also building integrations into many testing tools and they could easily help visualize the standard format you are publishing.

**Comment by Ma11hewThomas** [comment]
Thank you! I'm pleased to hear the GitHub plugin is helping your team. 

Following your suggestion we have built an additional method to [display historic test](https://github.com/ctrf-io/github-actions-test-reporter-ctrf?tab=readme-ov-file#historical-test-table) information. 

If you or your team have more ideas about how we can visualise historical test data in the GitHub Summary, please share with us. 

We also love the allure report!
